Bottom: 452f24543ea149615eb95f65920f0d47148192bb
Top:    c30f05667d8db1a54af542eb09f555d96866a3e1
Author: Johannes Rudolph <johannes_rudolph@gmx.de>
Date:   2009-10-05 22:27:38 +0200

split Local interface in two: ROLocal[+T] with load and Local[T] extends ROLocal[T] with store


---

diff --git a/bytecode/src/main/scala/net/virtualvoid/bytecode/Bytecode.scala b/bytecode/src/main/scala/net/virtualvoid/bytecode/Bytecode.scala
index e42c5d3..448e039 100644
--- a/bytecode/src/main/scala/net/virtualvoid/bytecode/Bytecode.scala
+++ b/bytecode/src/main/scala/net/virtualvoid/bytecode/Bytecode.scala
@@ -109,8 +109,10 @@ object Bytecode{
     											  ,thenB:F[R]=>Nothing):F[R]
   }
   
-  trait Local[T]{
-    def load[ST<:List]:F[ST] => F[ST**T]
+  trait ROLocal[+T]{
+    def load[R<:List,T2>:T]:F[R] => F[R**T2]
+  }  
+  trait Local[T] extends ROLocal[T]{
     def store[ST<:List]:F[ST**T] => F[ST]
   }
